examples/data_source/main.tf

```
@@ -13,20 +13,27 @@ provider "aap" {
   insecure_skip_verify = true
 }
 
-variable "state_id" {
-  type = number
-  description = "The id of the state to request"
+resource "aap_inventory" "my_inventory" {
+  name = "My new inventory"
+  description = "A new inventory for testing"
+  variables = jsonencode(
+    {
+      "foo": "bar"
+    }
+  )
 }
 
-data "aap_inventory" "sample" {
-  path = var.state_id
+output "inventory" {
+  value = aap_inventory.my_inventory
 }
 
-output "inventory_hosts" {
-  value = data.aap_inventory.sample.hosts
+data "aap_inventory" "sample" {
+  id = aap_inventory.my_inventory.id
```

```suggestion
  id = var.inventory_id
```
combined with the suggestion above